backend/src/realtime/realtime-note/realtime-note.ts

```diff
@@ -28,12 +28,13 @@ export class RealtimeNote extends EventEmitter2<RealtimeNoteEventMap> {
   private readonly clients = new Set<RealtimeConnection>();
   private isClosing = false;
 
-  constructor(private readonly note: Note, initialContent: string) {
+  constructor(private readonly note: Note, initialState: string | number[]) {
```

Comment:
Maybe we could use multiple contructors for those two different cases?

Reply:
Not possible in typescript. The way to go here is a union type.

Reply:
You could add two separate static methods to the class as factory methods